aws/provider_test.go

```diff
@@ -180,6 +181,27 @@ func testAccMatchResourceAttrRegionalARNNoAccount(resourceName, attributeName, a
 	}
 }
 
+// testAccMatchResourceAttrRegionalARNAccountID ensures the Terraform state regexp matches a formatted ARN with region and specific account ID
+func testAccMatchResourceAttrRegionalARNAccountID(resourceName, attributeName, accountID, arnService string, arnResourceRegexp *regexp.Regexp) resource.TestCheckFunc {
```

Comment:
😭 Please don't hate me, but I have the smallest nitpicks/questions:

- Since ARNs are in `arn:PARTITION:SERVICE:REGION:ACCOUNT:RESOURCE` format, could we have accountID be the second to last argument?
- Should we enforce a `*regexp.Regexp` type for the accountID argument if the intent is to use strings like `\d{12}`? Otherwise I wonder if we should just default that value in the `arnRegexp`, remove the argument, and name the function something like `testAccMatchResourceAttrRegionalAnyAccountARN()`? I think 12 digit account numbers and the `aws` string are the only expected values, unless there might be a test where we fully know the 12 digit account number, but I cannot think of a case like that anywhere.

Reply:
1. Yeah, that bugged me a bit when I did it. There was a reason, IIRC, but I'll change it
1. There's also at least `lambda` as an account ID. In this case, I was piggybacking on how `testAccMatch*ARN*()` build the final regex for the match. In the `*_service_account` data sources, we know the expected account ID, but those are `Check` not `Match` tests.

Reply:
Cool cool thanks!